lib/Core/DocumentPrioritizer/SimilarityResultPrioritizer.php

```diff
@@ -32,6 +32,8 @@ public function priority(?TextDocumentUri $one, ?TextDocumentUri $two): int
 
         $range = Suggestion::PRIORITY_LOW - Suggestion::PRIORITY_MEDIUM;
 
-        return Suggestion::PRIORITY_MEDIUM + $range - $range * $similarity;
+        $priority = (int) Suggestion::PRIORITY_MEDIUM + $range - $range * $similarity;
+
+        return $priority;
```

Comment:
I still don't get why phpstan is complaining.
The result was already cast to an integer thanks to the return type, we are not in strict mode so it should automatically cast without issue...
Any idea ?

Reply:
looks like a phpstan bug - does it work with the previous version of the code?

Reply:
actually I think it's a precedence issue (unless it's a precedence issue with phpstan) if we parenthesize it's fine: https://phpstan.org/r/26baabc8-fe48-412c-af02-2776266bbe76